### PR TITLE
Handle some edge cases in scores function

### PR DESCRIPTION
**Pull request by zmoon**
@btang1 pointed these out. These occur when all the data from either mod or obs or both is either all inside the interval or all outside.

* [x] add a test or two

cc: @btang1 @drnimbusrain

### CODE REVIEW

File: monet/util/stats.py [hunk 1]
```diff
@@ -1774,39 +1773,63 @@ def CSI(obs, mod, minval, maxval):
...
-    ct = crosstab(
+    import pandas as pd
+
+    df = pd.DataFrame({"obs": obs, "mod": mod})
+
+    # If NaN is involved in a cond, it will be F, we want to skip those
+    df = df.dropna(subset=["obs", "mod"], how="any")
+
+    if df.empty:
+        return 0, 0, 0, 0
```
Comment (zmoon) [inline monet/util/stats.py]:
```suggestion
        zero = np.float64(0)
        return zero, zero, zero, zero
```
For consistency with the normal return (and the docstring says float)

**Comment by bbakernoaa** [comment]
@btang1 can you take a look at this.  Go through the tests and make sure that a b c and d are being calculated correctly.

**Comment by btang1** [comment]
> @btang1 can you take a look at this. Go through the tests and make sure that a b c and d are being calculated correctly.

will do~

**Comment by btang1** [comment]
@zmoon @bbakernoaa 
I run 3 tests on Zach new score code (published 03/14) and the results are all good
1) test one: run with test_score.py example code. all good.
2) test two: run with test_score.py but set minval larger than maximum of obs and model. result all good. this is the very case we met when threshold lager than boundary.
3) run Zach's new scores with CSI plot for AQM upgrade project. and new code results are consistent with previous version (zach Mar-10th code) result. you can find result here: https://drive.google.com/drive/folders/1foauKK28-L7gj_46j-GCGCBb-F8L5bA9?usp=share_link

4) I figure out why my previous for loops code error. I set 'b' and 'c' reversely. after set them back, for loops code result is consistent with Zach's new code. result here: https://drive.google.com/drive/folders/1N8bVs8eC9UchZJrKt14alsjHMrllP8bU?usp=share_link

In conclusion, I run some test and all proved that Zach new version is best and ready to be published~
